scalalib/src/mill/javalib/android/AndroidAppModule.scala

```diff
@@ -36,34 +36,118 @@ trait AndroidAppModule extends JavaModule {
+      jarFiles ++= os.walk(extractDir).filter(_.ext == "jar").map(PathRef(_))
+
+      // If the res folder exists, add it to the resource folders
+      val resFolder = extractDir / "res"
+      if (os.exists(resFolder)) {
+        resFolders ++= Seq(PathRef(resFolder))
+      }
+    }
+
+    // Return both jar files and resource folders
+    (jarFiles, resFolders)
+  }
+
+  /**
+   * Overrides the `resources` task to include resources from unpacked AAR files
+   *
+   * @return Combined sequence of original and filtered AAR resources.
+   */
+  def resources: T[Seq[PathRef]] = Task {
+    // Call the function to unpack AARs and get the jar and resource paths
+    val (_, resFolders) = androidUnpackArchives()
+
+    // Combine and return all resources
+    super.resources() ++ resFolders
+  }
+
+  /**
+   * Overrides the compile classpath to replace `.aar` files with the extracted
+   * `.jar` files.
+   *
+   * @return The updated classpath with `.jar` files only.
+   */
+  override def compileClasspath: T[Agg[PathRef]] = Task {
+    // Call the function to get jar files and resource paths
+    val (jarFiles, _) = androidUnpackArchives()
+    super.compileClasspath().filter(_.path.ext == "jar") ++ jarFiles
+  }
+
+  def linkOptions: T[Seq[String]] = Task {
+    Seq[String]("--auto-add-overlay")
+  }
+
+  /**
+   * Compiles and links Android resources using `aapt2`, generating the `R.java` file.
+   *
+   * @return A `PathRef` to the directory containing the generated `R.java`.
+   *
+   * For more details on the aapt2 tool, refer to:
    * [[https://developer.android.com/tools/aapt2 aapt Documentation]]
    */
   def androidResources: T[PathRef] = Task {
-    val genDir: os.Path = T.dest // Directory to store generated resources.
+    val compiledResDir: os.Path = Task.dest / "compiled" // Directory for compiled resources
+    val resourceDirs = resources().map(_.path).filter(os.exists) // Merge all resource directories
+    os.makeDir.all(compiledResDir)
+    val compiledZips = resourceDirs.map { resDir =>
+      // Use the last two components of the path as part of the filename
+      val lastTwoSegments = resDir.segments.toSeq.takeRight(2).mkString("-")
+      val outputZip = compiledResDir / s"$lastTwoSegments.zip"
+      os.call((androidSdkModule().aapt2Path().path, "compile", "--dir", resDir, "-o", outputZip))
+      outputZip
+    }
 
-    os.call(Seq(
-      androidSdkModule().aaptPath().path.toString, // Call aapt tool
-      "package",
-      "-f",
-      "-m",
-      "-J",
-      genDir.toString, // Generate R.java files
-      "-M",
-      androidManifest().path.toString, // Use AndroidManifest.xml
-      "-I",
-      androidSdkModule().androidJarPath().path.toString // Include Android SDK JAR
-    ))
+    // Partition compiledZips into resourceZip and libZips
+    // "resources" is used to differ the libs res from app res
+    val (resourceZip, libZips) = compiledZips.partition(_.toString.contains("resources"))
```

Comment:
This `toString.contains("resources")` is still unacceptably sloppy. Since we construct the paths ourselves above in `outputZip`, we should be able to be more precise in exactly how we determine something is a resource vs a library. At the very least we should be able to look at `resDir.segments.toSeq` and check individual path segments rather than stringifying the whole thing